corehq/apps/geospatial/es.py

```diff
@@ -131,3 +134,34 @@ def mid(lower, upper):
     """
     assert lower <= upper
     return ceil(lower + (upper - lower) / 2)
+
+
+def case_query_for_missing_geopoint_val(domain, geo_case_property, case_type=None, size=None, offset=0):
+    query = (
+        CaseSearchES()
+        .domain(domain)
+        .filter(_geopoint_value_missing_for_property(geo_case_property))
+    )
+    if case_type:
+        query = query.case_type(case_type)
+    query.sort('opened_on')
+    query.start(offset)
+    if size:
+        query = query.size(size)
+    return query
+
+
+def _geopoint_value_missing_for_property(geo_case_property_name):
+    """
+    Query to find docs with missing 'geopoint_value' for the given case property.
+    """
+    return queries.nested(
+        CASE_PROPERTIES_PATH,
+        queries.filtered(
+            queries.match_all(),
```

Comment:
for my own learning, what does this do?

Reply:
I didn't implement this function, however taking a look at the [ES documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-all-query.html) I see it's a simple filter for matching all documents. Looking at the docstring for [the match_all() function definition in HQ](https://github.com/dimagi/commcare-hq/blob/1762ab76bcfdb31ffcd19c8e1292207de0e0bf53/corehq/apps/es/queries.py#L56), it reads the following:

> No-op query used because a default must be specified

This is getting used in a few other places where `queries.filtered()` is being called, such as [here](https://github.com/dimagi/commcare-hq/blob/1762ab76bcfdb31ffcd19c8e1292207de0e0bf53/corehq/apps/es/case_search.py#L281). My guess is that since `queries.filtered()` requires a query and filter param,  `match_all()` simply provides a default for the query param which then gets filtered on.